src/rust/cpu/instructions_0f.rs

```diff
@@ -1839,7 +1915,60 @@ pub unsafe fn instr_F20F7C_reg(r1: i32, r2: i32) { instr_F20F7C(read_xmm128s(r1)
 pub unsafe fn instr_F20F7C_mem(addr: i32, r: i32) {
     instr_F20F7C(return_on_pagefault!(safe_read128s(addr)), r);
 }
-
+#[no_mangle]
+pub unsafe fn instr_DF(source: reg128, r: i32) {
+    // fisttp xmm, xmm/mem128
+    let destination = read_xmm128s(r);
+    let result = reg128 {
+        f32: [
+            destination.f32[0] + destination.f32[1],
+            destination.f32[2] + destination.f32[3],
+            source.f32[0] + source.f32[1],
+            source.f32[2] + source.f32[3],
+        ],
+    };
+    write_xmm_reg128(r, result);
+}
+pub unsafe fn instr_DF_reg(r1: i32, r2: i32) { instr_DF(read_xmm128s(r1), r2); }
+pub unsafe fn instr_DF_mem(addr: i32, r: i32) {
+    instr_DF(return_on_pagefault!(safe_read128s(addr)), r);
+}
+#[no_mangle]
+pub unsafe fn instr_DD(source: reg128, r: i32) {
+    // fisttp xmm, xmm/mem128
+    let destination = read_xmm128s(r);
+    let result = reg128 {
+        f32: [
+            destination.f32[0] + destination.f32[1],
+            destination.f32[2] + destination.f32[3],
+            source.f32[0] + source.f32[1],
+            source.f32[2] + source.f32[3],
+        ],
+    };
+    write_xmm_reg128(r, result);
+}
+pub unsafe fn instr_DD_reg(r1: i32, r2: i32) { instr_DD(read_xmm128s(r1), r2); }
+pub unsafe fn instr_DD_mem(addr: i32, r: i32) {
+    instr_DD(return_on_pagefault!(safe_read128s(addr)), r);
+}
```

Comment:
These are floating point instructions. They should be in instructions.rs. See for example fpu_fistm16p.